pkg/provisioner/controller.go

```diff
@@ -337,6 +318,43 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 		Parameters:        class.Parameters,
 	}
 
+	// Fill in basic necessary object bucket info
+	setBasicOBInfo := func() {
+		setObjectBucketName(ob, key)
+		ob.Spec.StorageClassName = obc.Spec.StorageClassName
+		if ob.Spec.ReclaimPolicy == nil || (ob.Spec.ReclaimPolicy != nil && *ob.Spec.ReclaimPolicy == corev1.PersistentVolumeReclaimPolicy("")) {
+			// Do not blindly overwrite the reclaim policy. The provisioner might have reason to
+			// specify a reclaim policy that is  different from the storage class.
+			ob.Spec.ReclaimPolicy = options.ReclaimPolicy
+		}
```

Comment:
@jeffvance what do you think about doing a check to set (or not set) the reclaim policy only if the Provision implementation hasn't already set it?

Doing it this way means that in the case where Provision fails but still returns an OB, the OB will still get the default reclaim policy from the storage class before cleanup is called on it. The only possible hangup being that if the Provision implementation normally wants to override the cleanup policy, if it does not set its desired cleanup policy before the failure, this lib's controller will set the value. I don't foresee this being much of an issue. If the Provision implementation wants to specify a cleanup policy but doesn't do so in all cases including error cases where it still returns an OB, I view that as a bug in the Provision implementation.

Reply:
I accidentally clicked `Resolve` and can't figure out how to undo this.

Reply:
@BlaineEXE I thinks this is an improvement to handle a low probability use-case. Note that today the _reclaimPolicy_ does not influence lib cleanup, but it might in the future, and if that change were to be implemented then we'd want the `setBasicOBInfo` func to also check `err`.